examples/tutorials/core_api/5_profiling.py

```diff
@@ -0,0 +1,40 @@
+"""
+Stage 1: Let's create the core_context, and use it to start logging metrics.
+This is just a few lines of code, and we'll be able to see the results in the
+Determined WebUI.
+"""
+
+import logging
+import sys
+import time
+
+# NEW: import determined
+import determined as det
+from determined import profiler
+
+def main(core_context, increment_by):
+    core_context.profiler.profiler.set_training(True)
+    for batch in range(100):
+        with core_context.profiler.profiler.record_timing("train_batch.backward"):
```

Comment:
The `profiler.profiler` snippet looks like we're leaking implementation details into public APIs.

If the current ProfilerAgent really is ready to be promoted to public API status, it should probably just _be_ the `core_context.profiler`, rather than an attribute of it.

If not, then `core_context.profiler` should probably be a [facade pattern](https://en.wikipedia.org/wiki/Facade_pattern) object, exposing exactly the public API we want to expose from the underlying ProfilerAgent.

Reply:
i totally agree and meant to do this. updated.